pkg/lightning/backend/kv/context.go

```diff
@@ -0,0 +1,259 @@
+	return ctx.assertionLevel
+}
+
+// EnableMutationChecker implements the `table.MutateContext` interface.
+func (ctx *litTableMutateContext) EnableMutationChecker() bool {
+	return ctx.enableMutationChecker
+}
+
+// GetRowEncodingConfig implements the `table.MutateContext` interface.
+func (ctx *litTableMutateContext) GetRowEncodingConfig() tbctx.RowEncodingConfig {
+	return ctx.encodingConfig
+}
+
+// GetMutateBuffers implements the `table.MutateContext` interface.
+func (ctx *litTableMutateContext) GetMutateBuffers() *tbctx.MutateBuffers {
+	return ctx.mutateBuffers
+}
+
+// GetRowIDShardGenerator implements the `table.MutateContext` interface.
+func (ctx *litTableMutateContext) GetRowIDShardGenerator() *variable.RowIDShardGenerator {
+	return ctx.shardID
+}
+
+// GetReservedRowIDAlloc implements the `table.MutateContext` interface.
+func (ctx *litTableMutateContext) GetReservedRowIDAlloc() (*stmtctx.ReservedRowIDAlloc, bool) {
+	return &ctx.reservedRowIDAlloc, true
+}
+
+// GetBinlogSupport implements the `table.MutateContext` interface.
+func (*litTableMutateContext) GetBinlogSupport() (tbctx.BinlogSupport, bool) {
+	// lightning import does not support binlog.
+	return nil, false
+}
+
+// GetStatisticsSupport implements the `table.MutateContext` interface.
+func (ctx *litTableMutateContext) GetStatisticsSupport() (tbctx.StatisticsSupport, bool) {
+	return ctx, true
+}
+
+// UpdatePhysicalTableDelta implements the `table.StatisticsSupport` interface.
+func (ctx *litTableMutateContext) UpdatePhysicalTableDelta(
+	physicalTableID int64, delta int64,
+	count int64, cols variable.DeltaCols,
+) {
+	ctx.tableDelta.Lock()
+	defer ctx.tableDelta.Unlock()
+	if ctx.tableDelta.m == nil {
+		ctx.tableDelta.m = make(map[int64]variable.TableDelta)
+	}
+	tableMap := ctx.tableDelta.m
+	item := tableMap[physicalTableID]
+	item.Delta += delta
+	item.Count += count
+	item.TableID = physicalTableID
+	if cols != nil {
+		item.ColSize = cols.UpdateColSizeMap(item.ColSize)
+	}
+	tableMap[physicalTableID] = item
+}
+
+// GetColumnSize returns the colum size map (colID -> deltaSize) for the given table ID.
+func (ctx *litTableMutateContext) GetColumnSize(tblID int64) (ret map[int64]int64) {
+	ctx.tableDelta.Lock()
+	defer ctx.tableDelta.Unlock()
+	return maps.Clone(ctx.tableDelta.m[tblID].ColSize)
+}
+
+// GetCachedTableSupport implements the `table.MutateContext` interface.
+func (*litTableMutateContext) GetCachedTableSupport() (tbctx.CachedTableSupport, bool) {
+	// lightning import does not support cached table.
+	return nil, false
+}
+
+func (*litTableMutateContext) GetTemporaryTableSupport() (tbctx.TemporaryTableSupport, bool) {
+	// lightning import does not support temporary table.
+	return nil, false
+}
+
+func (*litTableMutateContext) GetExchangePartitionDMLSupport() (tbctx.ExchangePartitionDMLSupport, bool) {
+	// lightning import is not in a DML query, we do not need to support it.
+	return nil, false
+}
```

Comment:
PTAL @lance6716 @D3Hunter . some features not supported in lighthing/import into